pyvista/themes.py

```diff
@@ -197,7 +196,11 @@ def to_dict(self) -> dict:
         return dict_
 
     def __eq__(self, other):
-        if not isinstance(self, type(other)):
+        if not isinstance(other, _ThemeConfig):
+            return False
+
+        if not isinstance(self, Theme) and not isinstance(self, type(other)):
+            # Makes sure subconfigs are same type
```

Comment:
Do we need the subtype check? Wouldn't the different attributes fail equality checking anyway? Or, conversely: if two different subclasses actually have all the same parameters, aren't they equal?

I'm also always wary about `isinstance(self, type(other))` vs `isinstance(other, type(self))` vs `return NotImplemented`. Which is just to say that this is a treacherous piece of logic to touch here.

For instance, it looks like we posit `DocumentTheme() != DarkTheme()`, but we allow `DocumentTheme() == Theme()` (given modified parameters) based on type? I haven't tested this.

Reply:
> Do we need the subtype check?

We don't *need* it, so i removed it

> Wouldn't the different attributes fail equality checking anyway?

This is perfectly sufficient

> conversely: if two different subclasses actually have all the same parameters, aren't they equal?

Theis is what this block disallows but there's no good reason for this

I've removed this check

Reply:
The original comment might well have been different, but my comment is at least very related. 
 The current code has an awkward architecture, which creates these equality checking subtleties.  This method shows that we should consider two config classes, one `_ThemeConfig` and another is `_ThemeAttrConfig`.  Then the `__eq__` of each class would only check that `isinstance(other, <the appropriate class here>)`.  This _might_ also help clean up the tuple handling below, but I haven't looked closely.

Reply:
Should I implement `_ThemeAttrConfig` to do this?

Reply:
I think my comment was aimed mostly at the original implementation, sorry for not checking the new one more closely.  The current status is good with me.  My suggestion would apply if you wanted to put back in the original check that you had.  There _might_ be a problem if someone compares a `Theme` to a `_LightingConfig` object for example, but this issue can be addressed separately by adjusting the attribute checks.

Reply:
> there shouldn't be an external usage of `_LightingConfig` or similar config classes.

You have more confidence than I that users won't be doing unintended things :).  Users could be trying to mix and match Themes, for example:

```py
axes = DarkTheme.axes
my_axes_theme.axes = axes
# accidentally use `axes` somewhere where the intention was to use `my_axes_theme`

```

But let's leave it out of this PR IMO.

Reply:
I mainly think only a small group of users even know about themes in general but I really have no idea... custom theme support was only recently added.

Ineed much left to potentially address that could come later

Reply:
I feel this is enough of an edge case that we can either not address it or address it in another PR.